time-macros/src/lib.rs

```diff
@@ -89,3 +91,109 @@ pub fn format_description(input: TokenStream) -> TokenStream {
         DESCRIPTION
     }}
 }
+
+fn make_serde_serializer_module(
+    mod_name: proc_macro::Ident,
+    items: impl to_tokens::ToTokens,
+    formattable: TokenStream,
+) -> TokenStream {
+    let serialize_fns = quote! {
+        pub fn serialize<S: ::serde::Serializer>(
+            datetime: &#(formattable.clone()),
+            serializer: S,
+        ) -> Result<S::Ok, S::Error> {
+            use ::serde::Serialize;
+            datetime
+                .format(&DESCRIPTION)
+                .map_err(::time::error::Format::into_invalid_serde_value::<S>)?
+                .serialize(serializer)
+        }
+
+        pub fn deserialize<'a, D: ::serde::Deserializer<'a>>(
+            deserializer: D
+        ) -> Result<#(formattable.clone()), D::Error> {
+            use ::serde::Deserialize;
+            #(formattable.clone())::parse(<&str>::deserialize(deserializer)?, &DESCRIPTION)
+                .map_err(::time::error::Parse::to_invalid_serde_value::<D>)
+        }
+    };
+    let option_serialize_fns = quote! {
+        pub fn serialize<S: ::serde::Serializer>(
+            option: &Option<#(formattable.clone())>,
+            serializer: S,
+        ) -> Result<S::Ok, S::Error> {
+            use ::serde::Serialize;
+            option.map(|datetime| datetime.format(&DESCRIPTION))
+                    .transpose()
+                    .map_err(::time::error::Format::into_invalid_serde_value::<S>)?
+                    .serialize(serializer)
+        }
+
+        pub fn deserialize<'a, D: ::serde::Deserializer<'a>>(
+            deserializer: D
+        ) -> Result<Option<#(formattable.clone())>, D::Error> {
+            use ::serde::Deserialize;
+            Option::<&str>::deserialize(deserializer)?
+                .map(|string| #(formattable.clone())::parse(string, &DESCRIPTION))
+                .transpose()
+                .map_err(::time::error::Parse::to_invalid_serde_value::<D>)
+        }
+    };
+
+    quote! {
+        mod #(mod_name) {
+            use ::time::#(formattable.clone());
```

Comment:
No need to change this; I'm adding this as a note for myself to think about.

Do we want the implicit import or do we want to make the user specify the path if it's not in scope? The parser would need to be slightly adjusted to accept a path instead of an identifier.